x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/structurefinder/TimestampFormatFinder.java

```diff
@@ -288,6 +289,43 @@ public final class TimestampFormatFinder {
         )
     );
 
+    /**
+     * The first match in this list will be chosen, so it needs to be ordered
+     * such that more generic patterns come after more specific patterns.
+     */
+    static final List<CandidateTimestampFormat> ORDERED_CANDIDATE_FORMATS_ECS_V1;
+    static {
+        List<CandidateTimestampFormat> items = new ArrayList<>();
+        // CATALINA8_DATESTAMP %{MONTHDAY}-%{MONTH}-%{YEAR} %{HOUR}:%{MINUTE}:%{SECOND}
+        // 29-Aug-2009 12:03:33
```

Comment:
```suggestion
        // 29-Aug-2021 12:03:33.578
```

According to the example in the pattern file this always has millisecond accuracy.